### PR TITLE
Annotations: do not show fields mappings for -- grafana -- datasource

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes https://github.com/grafana/grafana/issues/51252

This PR hides the field mapping UI from the "-- Grafana --" editor

NewSectionMappingAnnotation

### CODE REVIEW

**Review by axelavargas — APPROVED** [review]
👏🏾 works as expected!

**Review by axelavargas — CHANGES_REQUESTED** [review]
Oh wait :(, I think there is a problem with other datasource annotations 🤔 , 

 I tested Prometheus datasource annotations and got this


AfterPR

but on Grafana play I have this

CurrentBehaviour

**Comment by ryantxu** [comment]
@axelavargas -- dooh, good catch!

This puts back the same check we had before #49858
image

It is odd, since the "type" is messed up... (should be id)

File: public/app/features/annotations/standardAnnotationSupport.ts [hunk 1]
```diff
@@ -231,13 +231,21 @@ export function getAnnotationsFromData(
...
+
 /**
  * Opt out of using the default mapping functionality on frontend.
  */
 export function shouldUseMappingUI(datasource: DataSourceApi): boolean {
   const { type } = datasource;
-  return (
-    type !== 'prometheus' && type !== 'elasticsearch' && type !== 'loki' && type !== 'grafana-opensearch-datasource'
+  return !(
+    type === 'datasource' || //  ODD behavior for "-- Grafana --" datasource
```
Comment (aocenas) [inline public/app/features/annotations/standardAnnotationSupport.ts:247]: @ashharrison90 so now we know why this was [here](https://github.com/grafana/grafana/pull/48814/files#diff-1c32b50f4e6010b94cfb26f3c99f93a48a0afae330294d206a9eeaa9393b53abL181) before.

**Review by aocenas — APPROVED** [review]

**Review by axelavargas — APPROVED** [review]

**Pull request merged**